### PR TITLE
[IMP] l10n_au: deferred GST improvement

### PR DESCRIPTION
**Pull request by vin-odoo**
Improve recording of DGST taxes by adding
a new tax for it, correcting the bas report
and adding a slight qol improvement when
registering dgst taxes via a misc entry.

task id # 3903331

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by vin-odoo** [comment]
see also https://github.com/odoo/enterprise/pull/69713

File: addons/l10n_au/models/account_move.py [hunk 1]
```diff
@@ -9,3 +9,22 @@ def _get_name_invoice_report(self):
...
             return 'l10n_au.report_invoice_document'
         return super()._get_name_invoice_report()
+
+    def _get_automatic_balancing_account(self):
+        """ Override to manage the DGST use case.
+        We want the automatic line to balance the DGST account to itself, as we only want the tax lines to have a real
+        impact.
+        """
+        # OVERRIDE account
+
```
Comment (FlorianGilbert) [inline addons/l10n_au/models/account_move.py:19]: You need a `self.ensure_one()` as you do the super at the end of this function

**Review by FlorianGilbert — COMMENTED** [review]
Small things
thank you

**Review by FlorianGilbert — APPROVED** [review]
@robodoo r+

**Comment by FlorianGilbert** [comment]
@robodoo r+

**Pull request closed without merging**